baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -127,6 +176,61 @@ def handle(self, message: kombu.Message) -> None:
             message.ack()
 
 
+class KombuBatchMessageHandler(MessageHandler):
+    def __init__(
+        self,
+        baseplate: Baseplate,
+        name: str,
+        handler_fn: BatchHandler,
+        error_handler_fn: Optional[BatchErrorHandler] = None,
+    ):
+        self.baseplate = baseplate
+        self.name = name
+        self.handler_fn = handler_fn
+        self.error_handler_fn = error_handler_fn
+
+    def handle(self, messages: Sequence[kombu.Message]) -> None:  # pylint: disable=arguments-differ
+        logger.info("Processing batch with %i messages", len(messages))
+
+        context = self.baseplate.make_context_object()
+        try:
+            # We place the call to ``baseplate.make_server_span`` inside the
+            # try/except block because we still want Baseplate to see and
+            # handle the error (publish it to error reporting)
+            with self.baseplate.make_server_span(context, self.name) as span:
+                span.set_tag("kind", "batch_consumer")
+
+                message: kombu.Message
+                for message in messages:
+                    delivery_info = message.delivery_info
+                    span.set_tag("amqp.routing_key", delivery_info.get("routing_key", ""))
```

Comment:
I think this is just going to override the tag each time you set this, so it's just going to be set to whatever it is set to in the last message.  I'm also guessing that these values are often going to be the same for each message in a batch or at least there will be significant overlap? It might be better to accumulate all of the unique patterns of these and just set each unique one as tags like below:

```python
class _SpanTag(NamedTuple):
    routing_key: str
    consumer_tag: str
    delivery_tag: str
    exchange: str


tags: Set[_SpanTag] = set()
for message in messages:
    delivery_info = message.delivery_info
    tags.add(_SpanTag(
        routing_key=delivery_info.get("routing_key", ""),
        consumer_tag=delivery_info.get("consumer_tag", ""),
        delivery_tag=delivery_info.get("delivery_tag", ""),
        exchange=delivery_info.get("exchange", ""),
    ))

for i, tag in enumerate(tags):
    span.set_tag(f"amqp.routing_key{i}", tag.routing_key)
    span.set_tag(f"amqp.consumer_tag{i}", tag.consumer_tag)
    span.set_tag(f"amqp.delivery_tag{i}", tag.delivery_tag)
    span.set_tag(f"amqp.exchange{i}", tag.exchange)
```